src/API/OnboardingTasks.php

```diff
@@ -878,29 +835,10 @@ public function snooze_task( $request ) {
 	 * @return WP_REST_Request|WP_Error
 	 */
 	public function undo_snooze_task( $request ) {
-		$id = $request->get_param( 'id' );
-
-		$snoozed = get_option( 'woocommerce_task_list_remind_me_later_tasks', array() );
-
-		if ( ! isset( $snoozed[ $id ] ) ) {
-			return new \WP_Error(
-				'woocommerce_tasks_invalid_task',
-				__( 'Sorry, no snoozed task with that ID was found.', 'woocommerce-admin' ),
-				array(
-					'status' => 404,
-				)
-			);
-		}
-
-		unset( $snoozed[ $id ] );
-
-		$update = update_option( 'woocommerce_task_list_remind_me_later_tasks', $snoozed );
-
-		if ( $update ) {
-			wc_admin_record_tracks_event( 'tasklist_undo_remindmelater_task', array( 'task_name' => $id ) );
-		}
-
-		return rest_ensure_response( TaskLists::get_task( $id ) );
+		$id   = $request->get_param( 'id' );
+		$task = TaskLists::get_task( $id );
```

Comment:
I think you'll still want to handle the case that the task ID is invalid, and this returns null. Right now it throws an unhandled error.

Reply:
Good call, though heads up that @louwie17 will probably be temporarily overriding this logic for backwards compatibility.